server/valgrind-out.txt

```diff
@@ -1,114 +1,166 @@
+--2570883-- REDIR: 0x4935b00 (libc.so.6:__strcpy_sse2_unaligned) redirected to 0x483f090 (strcpy)
+==2570883== Invalid read of size 8
+==2570883==    at 0x10B366: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883==  Address 0x4a780a0 is 192 bytes inside a block of size 200 free'd
+==2570883==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==2570883==    by 0x10B35E: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883==  Block was alloc'd at
+==2570883==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==2570883==    by 0x10B0E6: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883== 
+==2570883== 
+==2570883== HEAP SUMMARY:
+==2570883==     in use at exit: 744 bytes in 3 blocks
+==2570883==   total heap usage: 143 allocs, 140 frees, 327,099 bytes allocated
+==2570883== 
+==2570883== Searching for pointers to 3 not-freed blocks
+==2570883== Checked 16,863,000 bytes
+==2570883== 
+==2570883== 200 bytes in 1 blocks are still reachable in loss record 1 of 3
+==2570883==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==2570883==    by 0x10B0E6: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883== 
+==2570883== 272 bytes in 1 blocks are possibly lost in loss record 2 of 3
+==2570883==    at 0x483DD99: calloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==2570883==    by 0x40149DA: allocate_dtv (dl-tls.c:286)
+==2570883==    by 0x40149DA: _dl_allocate_tls (dl-tls.c:532)
+==2570883==    by 0x4865322: allocate_stack (allocatestack.c:622)
+==2570883==    by 0x4865322: pthread_create@@GLIBC_2.2.5 (pthread_create.c:660)
+==2570883==    by 0x10B071: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883== 
+==2570883== 272 bytes in 1 blocks are possibly lost in loss record 3 of 3
+==2570883==    at 0x483DD99: calloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==2570883==    by 0x40149DA: allocate_dtv (dl-tls.c:286)
+==2570883==    by 0x40149DA: _dl_allocate_tls (dl-tls.c:532)
+==2570883==    by 0x4865322: allocate_stack (allocatestack.c:622)
+==2570883==    by 0x4865322: pthread_create@@GLIBC_2.2.5 (pthread_create.c:660)
+==2570883==    by 0x10B23B: main (in /home/ashwinhr186/AESD/assignment-1-ashwinhr186/server/aesdsocket)
+==2570883== 
+==2570883== LEAK SUMMARY:
+==2570883==    definitely lost: 0 bytes in 0 blocks
+==2570883==    indirectly lost: 0 bytes in 0 blocks
+==2570883==      possibly lost: 544 bytes in 2 blocks
+==2570883==    still reachable: 200 bytes in 1 blocks
+==2570883==         suppressed: 0 bytes in 0 blocks
+==2570883== 
+==2570883== ERROR SUMMARY: 16 errors from 1 contexts (suppressed: 0 from 0)
+==2570883== 
+==2570883== 16 errors in context 1 of 1:
```

Comment:
@ashwinhr186 these are likely pointing to real problems.  Build with `-g` to help find them.